R/pkg/R/deserialize.R

```diff
@@ -56,8 +56,10 @@ readTypedObject <- function(con, type) {
 
 readString <- function(con) {
   stringLen <- readInt(con)
-  string <- readBin(con, raw(), stringLen, endian = "big")
-  rawToChar(string)
+  raw <- readBin(con, raw(), stringLen, endian = "big")
+  string <- rawToChar(raw)
+  Encoding(string) <- "UTF-8"
+  enc2native(string)
```

Comment:
I suspect I made a mistake here. It semms that enc2native just change the encoding mark, but does not convert the character from current encoding to the native encoding (iconv() can do this). As described in https://cran.r-project.org/doc/manuals/r-release/R-ints.html#Encodings-for-CHARSXPs, R can encodings other than that of the current locale, and most of the character manipulation functions now preserve UTF-8 encodings, we can keep the string encoding as "UTF-8". So could you try to remove this?